### ISSUE TITLE
Swagger Editor

### ISSUE THREAD

**Issue by sillyoldman** [micronaut-projects/micronaut-openapi#55]
Can I ask , if one also has a requirement to package a Swagger Editor with the application then if there's a recommended way? e.g. something like springfox ?

**Comment by puneetbehl** [comment]
I am guessing you might mean Swagger UI and not the Editor. Right now, micronaut-openapi doesn't package Swagger Editor or Swagger UI but if you want Swagger UI then you can use the following Gradle plugin:

https://github.com/int128/gradle-swagger-generator-plugin

**Comment by matthijsbierman** [comment]
Simplest way is to include an HTML page using [static resources support](https://docs.micronaut.io/latest/guide/index.html#staticResources).

In `application.yml`:

```yaml
micronaut:
  router:
    static-resources:
      swagger:
        paths:
          - classpath:META-INF/swagger
          - classpath:swagger
        mapping: /swagger/**
```

And in `src/main/swagger` put an `index.html` with:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>OpenAPI documentation</title>
    <link rel="icon" type="image/png" href="https://unpkg.com/swagger-ui-dist@3.23.0/favicon-32x32.png">
    <link rel="stylesheet" type="text/css" href="https://unpkg.com/swagger-ui-dist@3.23.0/swagger-ui.css">
    <script src="//unpkg.com/swagger-ui-dist@3.23.0/swagger-ui-standalone-preset.js"></script>
    <script src="//unpkg.com/swagger-ui-dist@3.23.0/swagger-ui-bundle.js"></script>
</head>
<body>

<div id="swagger-ui"></div>
<script>
    window.onload = function() {
        var ui = SwaggerUIBundle({
            urls: [{
                name: "API documentation",
                url: "/swagger/swagger.yml" // TODO may be different
            }],
            dom_id: '#swagger-ui',
            deepLinking: true,
            presets: [
                SwaggerUIBundle.presets.apis,
                SwaggerUIStandalonePreset
            ],
            plugins: [
                SwaggerUIBundle.plugins.DownloadUrl
            ],
            layout: "StandaloneLayout"
        });
        window.ui = ui;
    }
</script>
</body>
</html>
```

The path to the YAML file is hardcoded. It can be set by an environment variable (micronaut.openapi.target.file) or generated from `@OpenAPIDefinition`. If none of those are available then it falls back to `swagger.yml`.

When the project is compiled it shows it in the logs:
```Writing OpenAPI YAML to destination: ```

Now you can access Swagger UI at http://localhost:8080/swagger/.

**Comment by nkatakkar** [comment]
@matthijsbierman Thank you. Will try using the static resources route.

Also Thank you @puneetbehl  
Since we use Maven, will try build tool agnostic approach first.

Thanks.

**Comment by sillyoldman** [comment]
@matthijsbierman Worked very well! Cheers.

**Comment by matthijsbierman** [comment]
Note: there is now a built-in way to expose swagger-ui:

https://micronaut-projects.github.io/micronaut-openapi/1.3.x/guide/index.html#openApiViews

**Comment by tweiand** [comment]
@matthijsbierman - the provided latest documentation does not work. That is why we are looking for other ways.

**Comment by matthijsbierman** [comment]
@tweiand the built-in way is not officially released yet, it will be available with 1.3.0. If you want to try out the milestone release you can install with:
```sdk install micronaut 1.3.0.M1```
